src/ui/common/form/SwitchRenderer.js

```diff
@@ -2,29 +2,34 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { Switch, Col, ControlLabel } from 'patternfly-react';
 
-const switchField = (input, switchValue, trueValue, falseValue, onToggleValue) => {
+const switchField = (input, switchValue, trueValue, falseValue, onToggleValue, dataTestId) => {
   const handleChange = (el, val) => {
     const returnVal = val ? trueValue : falseValue;
     input.onChange(returnVal);
     if (onToggleValue) {
       onToggleValue(returnVal);
     }
   };
-
-  return (
-    <Switch
-      {...input}
-      value={switchValue}
-      onChange={handleChange}
-    />
-  );
+  const switchComponent = (<Switch
+    {...input}
+    value={switchValue}
+    onChange={handleChange}
+  />);
+  if (dataTestId) {
```

Comment:
I'd avoid this coupling of dataTestId with component logic. Even a wrapping div with data-testid is better.

Reply:
if removed. Thanks